### PR TITLE
Avoid unnecessary pyenv usage in environment setup

### LINKED ISSUE #26078: Avoid unnecessary pyenv usage in environment setup
### What's the issue or suggestion?

Going through https://docs.dagster.io/community/contributing, it seems pyenv is used to create an isolated environment, inside of which uv is installed. Since uv can also be used to create virtual environments, it makes sense to just install uv first (no Homebrew necessary!) and use it all the way. Realistically, all that changes in scripts/install_dev_python_modules.py is that the following lines can be removed:

```diff
-    # Ensure uv is installed which we use for faster package resolution
-    subprocess.run(["pip", "install", "-U", "uv"], check=True)
-
```

I think it would also require some (minimal) changes to python_modules/automation/automation/docker/images/buildkite-test/Dockerfile.

Happy to make a PR, if this makes sense!

### Additional information

_No response_

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

### PR DESCRIPTION
**Pull request by deepyaman**
## Summary & Motivation

Resolve #26078

## How I Tested These Changes

The change in instructions reflects how I set up my local environment.

I verified that I can build the Buildkite image using `dagster-image build --name buildkite-test  --dagster-version unused -v 3.12`, and that dependencies are installed on the resulting image, but the CI tests are not running using the updated image.

## Changelog

Avoided unnecessary pyenv usage in environment setup step of contributing guide.

### CODE REVIEW

**Comment by neverett** [comment]
Tagging @alangenfeld and @smackesey since this is a change to the core engineering development process and not strictly a docs change.

**Comment by deepyaman** [comment]
Hmm... this seems to pass by and large (`helm` failure seems unrelated), but I also haven't figured out yet where changes to python_modules/automation/automation/docker/images/buildkite-test/Dockerfile are actually getting used. The actual tests seem to be getting run via tox config generated in .buildkite/dagster-buildkite/dagster_buildkite/steps/tox.py.

**Review by schrockn — APPROVED** [review]
Great thank you for updating

File: scripts/install_dev_python_modules.py [hunk 1]
```diff
@@ -136,9 +133,6 @@ def main(
             "https://github.com/dagster-io/build-grpcio/wiki/Wheels",
         ]
 
-    # Ensure uv is installed which we use for faster package resolution
```
[the remark this answers is not in the document]
Comment (deepyaman) [reply scripts/install_dev_python_modules.py:139]: If somebody setting up a new environment (with the updated instructions) were to run into this, they'd end up using whatever `pip` was found to install `uv`, which I don't know would be ideal IMO.

If necessary, could add a separate check to ensure `uv` is available, but I think it's fine to just error down here if `uv` isn't found.

File: python_modules/automation/automation/docker/images/buildkite-test/Dockerfile [hunk 2]
```diff
@@ -131,8 +131,8 @@ RUN curl -fsSL https://download.docker.com/linux/debian/gpg | \
     && curl -L "https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose \
     && chmod +x /usr/local/bin/docker-compose
 
-# Install Python build deps
-RUN pip install -U pip setuptools wheel
+# Install uv
+COPY --from=ghcr.io/astral-sh/uv:0.5.4 /uv /bin/
```
Comment (deepyaman) [inline python_modules/automation/automation/docker/images/buildkite-test/Dockerfile:135]: I wasn't able to use a variable for the version (as I initially tried); I think the issue is related to https://github.com/moby/moby/issues/34482.

This is consistent with how things like the JDK are being installed.

File: python_modules/automation/automation/docker/images/buildkite-test/Dockerfile [hunk 3]
```diff
@@ -169,7 +168,7 @@ COPY --from=snapshot_builder /snapshot-reqs.txt .
 
 # Preinstall non-Dagster packages in image so that the virtual environment
 # builds faster in Buildkite.
-RUN uv pip install -r /snapshot-reqs.txt \
+RUN uv pip install --system -r /snapshot-reqs.txt \
```
Comment (schrockn) [inline python_modules/automation/automation/docker/images/buildkite-test/Dockerfile:171]: why is this change necessary?

File: python_modules/automation/automation/docker/images/buildkite-test/Dockerfile (hunk 2 above)
Comment (schrockn) [reply python_modules/automation/automation/docker/images/buildkite-test/Dockerfile:135]: So was this just not working before?

**Review by schrockn — APPROVED** [review]
Approving to unblock. Please respond to my comments for my own understanding. Thanks!

File: python_modules/automation/automation/docker/images/buildkite-test/Dockerfile (hunk 2 above)
Comment (deepyaman) [reply python_modules/automation/automation/docker/images/buildkite-test/Dockerfile:135]: Previously, the Dockerfile just upgraded pip, and `scripts/install_dev_python_modules.py` installed uv using pip. Since updating `scripts/install_dev_python_modules.py` to just use uv directly (instead of pip to install uv, and then using uv), uv should be installed directly.

File: python_modules/automation/automation/docker/images/buildkite-test/Dockerfile (hunk 3 above)
Comment (deepyaman) [reply python_modules/automation/automation/docker/images/buildkite-test/Dockerfile:171]: Unlike pip, uv doesn't allow installation without a virtual environment by default.

It should still be possible to `pip install` (instead of `uv pip install`), but imagine that would be unnecessarily slower, especially if everything else is using uv.
